op-challenger/fault/agent.go

```diff
@@ -65,10 +75,32 @@ func (a *Agent) Act(ctx context.Context) error {
 	return nil
 }
 
+// ShouldResolve returns true if the agent should resolve the game.
+func (a *Agent) ShouldResolve(ctx context.Context, status uint8) bool {
+	expectedStatus := types.GameStatusDefenderWon
+	if a.agreeWithProposedOutput {
+		expectedStatus = types.GameStatusChallengerWon
+	}
+	if expectedStatus != types.GameStatus(status) {
+		a.log.Warn(
+			"Not resolving game with disagreed status",
+			"expected_status",
+			expectedStatus,
+			"status",
+			status,
+		)
+	}
+	return expectedStatus == types.GameStatus(status)
+}
+
 // tryResolve resolves the game if it is in a terminal state
 // and returns true if the game resolves successfully.
 func (a *Agent) tryResolve(ctx context.Context) bool {
-	if !a.responder.CanResolve(ctx) {
+	status, err := a.responder.CallResolve(ctx)
+	if err != nil {
+		return false
+	}
+	if status != 0 && !a.ShouldResolve(ctx, status) {
```

Comment:
Why isn't this `status != 0` check inside `ShouldResolve`? It seems like `ShouldResovlve(tyep.GameStatusInProgress` should return false.

Reply:
Agreed, good callout.